### PR TITLE
Allow access to DatabaseError code

### PR DESCRIPTION
**Pull request by marvinkite**
When I try to implement the error handling I find it hard to know what kind of error the Driver returns. As  #142 also suggested some improvement I think this simply allows to know what kind of error my code should deal with. 

For instance the code could be `Neo.ClientError.Schema.ConstraintValidationFailed` but through the `error` interface I can access only the string message `Server error: [Neo.ClientError.Schema.ConstraintValidationFailed] Node(6) already exists with label `. I think accessing the Code directly rather than parsing the string improves the usability of the library. 

The error types are internal which makes the error handling impossible so I agree with the need to improve it some way. These utility methods could be one way I agree some types of error messages could be internal because the Client application don't need to be concerned but when it's related to the consequences of Cypher and the Parameters so the client app shell handle it _(Like Constraint or Security)_ the error could expose more than just a string message.

### CODE REVIEW

**Comment by 2hdddg** [comment]
Hi @marvinkite 
Thanks for your pull request, you need to sign the CLA for the PR to get merged.
The database error struct is planned to be moved from internal package to public in 4.0, future for these error check functions are not determined yet. The approach you take here on 1.8 looks nice but is probably not the way on 4.0 (which means another deprecated API).

File: neo4j/error.go [hunk 1]
```diff
@@ -94,3 +94,13 @@ func IsServiceUnavailable(err error) bool {
...
+
+// GetDatabaseErrorCode is a utility method to check if the provided error is DatabaseError and if so
+// it gets the error code.
+func GetDatabaseErrorCode(err error) string {
+	dbErr, is := err.(*db.DatabaseError)
+	if !is {
+		return ""
+	}
+	return dbErr.Code
+}
```
Comment (grizzly-monkey) [inline neo4j/error.go]: add eol.

Comment (marvinkite) [reply neo4j/error.go]: Fixed it and signed the CLA but I have no idea why the check fails.

Comment (2hdddg) [reply neo4j/error.go]: Still CLA problems

Comment (marvinkite) [reply neo4j/error.go]: Is there anything I can do to resolve the CLA problem? I followed the process to send the email. I cause could be the first commit does not contains the name or email but the second has the correct github username so I don't know how the check performed.

Comment (2hdddg) [reply neo4j/error.go]: The build needs to rerun to check again and it triggers on commits.

We are very thankful for pull request and this one is nicely done but 1.8 is not the focus right now. Can you please have a look on 4.0-dev branch and see if the work done there solves your problem. 4.0 is on it's way, hopefully alpha pretty soon.

**Comment by 2hdddg** [comment]
If you really need this functionality on 1.8 feel free to open again.

**Pull request closed without merging**